### PR TITLE
ODM-12113 download files

### PR DESCRIPTION
**Pull request by viapivov**
image

image

image

### CODE REVIEW

**Review by genestack-okunitsyn — DISMISSED** [review]

**Review by genestack-okunitsyn — DISMISSED** [review]

File: openapi/swagger/fs/usr/share/nginx/html/helper/index.html [hunk 1]
```diff
@@ -99,6 +99,9 @@ <h4>Query/retrieve data</h4>
           <li>
             <a href="/swagger/?urls.primaryName=flowCytometryUser">flowCytometryUser</a> - API endpoints for retrieving only flow cytometry data or metadata
           </li>
+          <li>
+              <a href="/swagger/?urls.primaryName=filesUser">filesUser</a> - API endpoints for retrieving files
```
Comment (m-smazhevsky) [inline openapi/swagger/fs/usr/share/nginx/html/helper/index.html]:
```suggestion
              <a href="/swagger/?urls.primaryName=filesUser">filesUser</a> - API endpoints for retrieving data or metadata of files, attached to a study
```
This is a my suggestion, but let's approve with product guys

File: openapi/swagger/fs/usr/share/nginx/html/helper/index.html [hunk 2]
```diff
@@ -139,6 +142,9 @@ <h4>Import/curate data</h4>
           <li>
             <a href="/swagger/?urls.primaryName=manageData">manageData</a> - API endpoints for managing data
           </li>
+          <li>
+              <a href="/swagger/?urls.primaryName=filesCurator">filesCurator</a> - API endpoints for managing files
+          </li>
```
Comment (m-smazhevsky) [inline openapi/swagger/fs/usr/share/nginx/html/helper/index.html]:
```suggestion
              <a href="/swagger/?urls.primaryName=filesCurator">filesCurator</a> - API endpoints for managing files, attached to a study
          </li>
```
This is a my suggestion, but let's approve with product guys

File: openapi/v1/filesUser.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+openapi: 3.1.0
+info:
+  description: |-
+    This swagger page describes the filesUser APIs.
```
Comment (m-smazhevsky) [inline openapi/v1/filesUser.yaml]: I think we need more details here

Comment (m-smazhevsky) [reply openapi/v1/filesUser.yaml]: same for filesCurator

**Review by m-smazhevsky — CHANGES_REQUESTED** [review]
Looks good, especially the new color :)

The file will always look as link to download in swagger? Or small files will be downloaded right away?

File: openapi/v1/filesCurator.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      operationId: headFileAsCurator
+      parameters:
+        - description: Unique identifier (accession) of the object.
+          in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
```
Comment (m-smazhevsky) [inline openapi/v1/filesCurator.yaml]: I don't know the best practices for head request, but maybe it should be 204?

Comment (viapivov) [reply openapi/v1/filesCurator.yaml]: image

Let's just copycat S3

File: openapi/v1/filesUser.yaml (hunk 3 above)
Comment (viapivov) [reply openapi/v1/filesUser.yaml]: I'm afraid we can not write documentation without seeing the whole picture

**Comment by viapivov** [comment]
This will always work as download-file

**Review by m-smazhevsky — DISMISSED** [review]

**Review by genestack-okunitsyn — APPROVED** [review]

**Pull request merged**